File: 프로그래머스/2/60057. 문자열 압축/문자열 압축.py
def solution(s):
    total_length = len(s)
    answer = min(map(lambda length: get_length(s, length), range(1, total_length//2 + 1)), default=1) 
    return answer

def get_length(s, length):
    total_length = len(s)
    bef, count, total = "", 1, 0
    tmp = []
    for idx in range(0, total_length, length):
        now = s[idx:idx+length]
        if bef == now:
            count += 1
        else:
            tmp.append(bef)
            if count != 1:
                tmp.append(count)
            total += len(bef) + (1 if count != 1 else 0)
            bef = now
            count = 1
    total += len(bef) + (1 if count != 1 else 0)
    tmp.append(bef)
    if count != 1:
        tmp.append(count)

    return len(''.join(map(str, tmp)))
    

File: 프로그래머스/2/60057. 문자열 압축/문자열 압축.java
public class Solution {
    public int solution(String s) {
        int totalLength = s.length();
        int answer = totalLength;

        for (int len = 1; len <= totalLength / 2; len++)
            answer = Math.min(answer, getLength(s, len));

        return answer;
    }

    private int getLength(String s, int len) {
        int totalLength = s.length();
        String prev = "";
        int count = 1;
        int nowLength = 0;

        for (int i = 0; i < totalLength; i += len) {
            String now = s.substring(i, Math.min(i + len, totalLength));

            if (now.equals(prev)) {
                count++;
            } else {
                nowLength += prev.length();
                if (count > 1) {
                    nowLength += String.valueOf(count).length();
                }
                prev = now;
                count = 1;
            }
        }

        nowLength += prev.length();
        if (count > 1) {
            nowLength += String.valueOf(count).length();
        }

        return nowLength;
    }
}